### PR TITLE
LPAL-526 Retrospective ADRs

### PR DESCRIPTION
**Pull request by townxelliot**
## Purpose

Fixes [LPAL-526](https://opgtransform.atlassian.net/browse/LPAL-526).

## Approach

Add Architecture Decision Records covering recent work.

## Learning

n/a

## Checklist

* [X] I have performed a self-review of my own code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] I have added mandatory tags to terraformed resources, where possible
* [ ] If I added a package.json or composer.json, I also made sure this is included in the script in `.github/workflows/dependabot-update.yml`
* [ ] The product team have tested these changes

### CODE REVIEW

**Comment by johnnolan** [comment]
Are you going to include the ethics side in these or would that come at a later date/decide whether to at a later date?

**Comment by townxelliot** [comment]
> Are you going to include the ethics side in these or would that come at a later date/decide whether to at a later date?

Could you give an example of what the ethics side would look like, please?

I did dither about including certain non-technical elements in these ADRs, as they're not strictly "architecture". In the end I did include some borderline-philosophical statements, insofar as they impact on technical decision making, so can see how ethics might fit in that bracket.

**Comment by townxelliot** [comment]
Comments raised by Amy Wilson which I will address:

* I will move the structurizr file out of the architecture directory. It's not really an image, so I may create a new "design" directory for it.
* I will add a note about the reasons for doing load testing, to the effect: while we have auto-scaling, we don't have a way to check that it functions correctly, or work out what the top end of our capacity actually is (we sort of guesstimate that it's enough); we also want a way to test that any new components we add can cope with the levels of traffic we know we get before they are put into production.

**Comment by johnnolan** [comment]
> > Are you going to include the ethics side in these or would that come at a later date/decide whether to at a later date?
> 
> Could you give an example of what the ethics side would look like, please?
> 
> I did dither about including certain non-technical elements in these ADRs, as they're not strictly "architecture". In the end I did include some borderline-philosophical statements, insofar as they impact on technical decision making, so can see how ethics might fit in that bracket.

Absolutely, on Modernising we are adding some extra things around ethics. A better explanation is here https://docs.modernising.opg.service.justice.gov.uk/research-development/articles/tech-ethic-decisions/ and a list of existing ones for examples are here https://docs.modernising.opg.service.justice.gov.uk/adr/

No pressure to do this, just more if you as a team want to include things like this moving forward where appropriate

**Review by williamfalconeruk — DISMISSED** [review]
Happy with this so far, can't see any concerns from a webops perspective. all of the points here we've discussed as well, and it's good to read about the thought process / context.
ADR 0006 is a really good manifesto for what we're trying to achieve on this legacy codebase, makes pretty clear where we're going with it. Good job!

**Review by nickdavis2001 — APPROVED** [review]
Looks good to me 🏆

**Comment by nickdavis2001** [comment]
There are other subjects we can gradually add beyond this, e:g more about testing strategy.

**Comment by townxelliot** [comment]
> There are other subjects we can gradually add beyond this, e:g more about testing strategy.

Good idea @nickdavis2001. I suggest you raise an issue for that ADR :)

File: docs/architecture/0006-modernise-the-code-base.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+We have decided to modernise the code base to make it easier to work with and better
+aligned with modern web architecture and standards. This is not a small job, but
+the guiding principles we've decided on, shown below, should help us achieve our aims.
+
+("Modernising the code base" is not to be confused with "modernising LPAs". Here
+we're just talking about modernising the code base for the Make an LPA tool.)
+
+* **Don't rewrite everything at once**
+
+  Where possible, migrate part of an application to a new
```
Comment (townxelliot) [inline docs/architecture/0006-modernise-the-code-base.md:44]: "to new"

File: docs/architecture/0006-modernise-the-code-base.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  components and split traffic coming into the domain so that some paths are diverted to that
```
Comment (townxelliot) [inline docs/architecture/0006-modernise-the-code-base.md:45]: "component" (singular)

**Comment by townxelliot** [comment]
> > > Are you going to include the ethics side in these or would that come at a later date/decide whether to at a later date?
> > 
> > 
> > Could you give an example of what the ethics side would look like, please?
> > I did dither about including certain non-technical elements in these ADRs, as they're not strictly "architecture". In the end I did include some borderline-philosophical statements, insofar as they impact on technical decision making, so can see how ethics might fit in that bracket.
> 
> Absolutely, on Modernising we are adding some extra things around ethics. A better explanation is here https://docs.modernising.opg.service.justice.gov.uk/research-development/articles/tech-ethic-decisions/ and a list of existing ones for examples are here https://docs.modernising.opg.service.justice.gov.uk/adr/
> 
> No pressure to do this, just more if you as a team want to include things like this moving forward where appropriate

Thanks @johnnolan, I think that's a good idea. I'm going to put in an issue to incorporate thinking about ethics more explicitly as part of our decision-making process, which is itself a decision :) The examples and explanation you provided are very helpful, too.

**Review by johnnolan — APPROVED** [review]

**Pull request merged**